src/gl/lineFeature.js

```diff
@@ -151,6 +152,19 @@ var gl_lineFeature = function (arg) {
         position.push(pos.x);
         position.push(pos.y);
         position.push(pos.z || 0.0);
+        if (!j) {
+          firstpos = pos;
+        }
+      }
+      if (lineItem.length > 2 && closedFunc(data[i], i)) {
```

Comment:
Our general method of getting style information from functions is to pass two parameters: the current data item and its index.